### PR TITLE
Documentation improvements

### PR DESCRIPTION
**Pull request by pbrisbin**
Please see individual commits for more details.

- Rewrite README
- Re-org modules
- Re-document extracted modules
- Re-order and re-document Graphula exports

To review the Haddocks locally, checkout this branch and run:

```console
stack --work-dir .stack-work-haddocks build --haddock --open
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1 @@
-test/README.lhs
```
Comment (pbrisbin) [inline README.md:1]: Oh, I totally missed that this was a symlink, just to a not-top-level file. I will re-un-correct this.

File: src/Graphula/Node.hs [hunk 2]
```diff
@@ -54,11 +56,9 @@ import UnliftIO.Exception (Exception, throwIO)
...
 -- and 'mempty'.
 --
--- @
--- a1 <- node @A () mempty
--- a2 <- node @A () $ edit $ \a -> a { someField = True }
--- a3 <- node @A () $ ensure $ (== True) . someField
--- @
+-- > a1 <- node @A () mempty
+-- > a2 <- node @A () $ edit $ \a -> a { someField = True }
+-- > a3 <- node @A () $ ensure $ (== True) . someField
```
Comment (pbrisbin) [inline src/Graphula/Node.hs:61]: I'm having a lot of trouble getting the `@`/`@` style code blocks (which support `'linking'` identifiers) to work reliably with `@TypeApplications` present. There doesn't seem to be a pattern to when they break or are parsed correctly, vanish, or result in breaking the rest of the line. So I'm resorting to `>`-style in a lot of places.

**Comment by eborden** [comment]
> To review the Haddocks locally, checkout this branch and run:
> ```
> stack --work-dir .stack-work-haddocks build --haddock --open
> ```

I wish there was some generalizable pattern here for reviewing built assets. Things I've wanted to review  the built asset for:
- Your recent blog post
- The haddocks here
- Literally like anything (hello ephemeral anythings)

**Review by eborden — APPROVED** [review]
Leaps and bounds better than before. Thanks for digging in this!

**Comment by pbrisbin** [comment]
I agree there should be a better way, but you at least know about Rich Diff in the blog post PR right?

**Pull request merged**